packages/odie-client/src/components/message/sources.tsx

```diff
@@ -12,6 +12,18 @@ export const Sources = ( { message }: { message: Message } ) => {
 	const sources = useMemo( () => {
 		const messageLength = message?.context?.sources?.length ?? 0;
 		if ( messageLength > 0 ) {
+			// Record TrainTracks render events
+			let uiPosition = 0;
+			message.context?.sources?.forEach( ( source: Source ) => {
+				trackEvent( 'sources_traintracks_render', {
+					fetch_algo: source?.railcar?.fetch_algo,
+					ui_algo: source?.railcar?.ui_algo,
+					railcar: source?.railcar?.railcar,
+					fetch_position: source?.railcar?.fetch_position,
+					ui_position: uiPosition,
+				} );
+				uiPosition++;
+			} );
```

Comment:
```suggestion
			message.context?.sources?.forEach( ( source: Source, index: number ) => {
				trackEvent( 'sources_traintracks_render', {
					fetch_algo: source?.railcar?.fetch_algo,
					ui_algo: source?.railcar?.ui_algo,
					railcar: source?.railcar?.railcar,
					fetch_position: source?.railcar?.fetch_position,
					ui_position: index,
				} );
				uiPosition++;
			} );
```

Reply:
Thanks for the suggestion Daniel. I updated it.